integ/src/main.rs

```diff
@@ -217,11 +217,14 @@ async fn run() -> Result<()> {
             )
             .await
             .context(error::LoadKubeConfigSnafu)?;
+
+            // infer the default namespace from the loaded kubeconfig
+            let namespace = config.default_namespace.to_string();
```

Comment:
If the user can configure the namespace of their choice, how is it that you get that from the kubeconfig?

Reply:
Good question: the `config.default_namespace` property is the namespace defined by the loaded client config. This is typically the namespace of where the execution is currently happening.

So, in the case of the controller, it reads the in-cluster config and infers that it is in the "bottlerocket-aws" (or whatever the namespace was set during deploy).

This case in the integration tests is abit different in that we aren't inferring a in-cluster client config, we load one explicitly from eksctl. But the principle is the same - it infers the default namespace to deploy some stuff to (and in this case, it is actually "default")

It's abit confusing because the name "default" is overloaded. Most of the time, from the controllers perspective, we are simply attempting to infer what namespace the current context is executing from.